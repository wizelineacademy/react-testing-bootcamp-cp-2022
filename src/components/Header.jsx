import React from "react";

const style = {
  background: "#A0BCC2",
  textAlign: "center",
  height: "20vh",
  maxWidth: "90vw",
  lineHeight: "20vh",
  margin: "auto",
  marginTop: "1vh"
};

export const Header = () => {
  return (
    <div style={style}>
      <h1>NASA - Picture of the Day</h1>
    </div>
  );
};
