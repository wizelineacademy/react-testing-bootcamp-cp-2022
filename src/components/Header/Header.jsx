import styled from "styled-components";

const Header = () => (
	<header>
		<Title>Nasa's picture of the day</Title>
	</header>
);

const Title = styled.h1`
	text-align: center;
`;

export default Header;
