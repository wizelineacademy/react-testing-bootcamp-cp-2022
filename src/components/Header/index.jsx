import { Container, Navbar } from 'react-bootstrap';

const Header = () => (
  <Navbar bg='light' variant='light' expand='lg'>
    <Container>
      <Navbar.Brand>
        <h2>Wizeline Testing Proyect</h2>
      </Navbar.Brand>
    </Container>
  </Navbar>
);

export default Header;
