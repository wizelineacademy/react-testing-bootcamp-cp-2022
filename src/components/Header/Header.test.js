import { render, screen } from "@testing-library/react";

import Header from ".";

const setUp = () => render(<Header />);

describe("Header", () => {
	it("should render the title", () => {
		const { container } = setUp();
		const title = screen.getByRole("heading", {
			name: /nasa's picture of the day/i,
		});

		expect(title).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});
});
