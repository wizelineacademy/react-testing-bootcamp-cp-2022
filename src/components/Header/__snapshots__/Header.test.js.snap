// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render the title 1`] = `
<div>
  <header>
    <h1
      class="sc-bczRLJ hAGsLM"
    >
      Nasa's picture of the day
    </h1>
  </header>
</div>
`;
